tests/models/musicgen/test_modeling_musicgen.py

```diff
@@ -189,6 +198,47 @@ def setUp(self):
     def test_config(self):
         self.config_tester.run_common_tests()
 
+    # special case for labels
+    def _prepare_for_class(self, inputs_dict, model_class, return_labels=False):
+        inputs_dict = super()._prepare_for_class(inputs_dict, model_class, return_labels=return_labels)
+
+        if return_labels:
+            inputs_dict["labels"] = torch.zeros(
+                (self.model_tester.batch_size, self.model_tester.seq_length, self.model_tester.num_codebooks),
+                dtype=torch.long,
+                device=torch_device,
+            )
+        else:
+            inputs_dict.pop("labels", None)
```

Comment:
Why would you need to pop if `return_labels` is passed in the `super()._prepare_for_class` call?